scripts/wildcard-domain-processor/domain-extractor.ts

```diff
@@ -0,0 +1,98 @@
+import agtree, {
+    AnyRule,
+    DomainListParser,
+    Modifier,
+} from '@adguard/agtree';
+import { utils } from './utils';
+
+/**
+ * A list of network rule modifiers that are to be scanned for dead domains.
```

Comment:
cosmetic rules can have `$domain` modifier as well
https://adguard.com/kb/general/ad-filtering/create-own-filters/#non-basic-domain-modifier

btw there are no tests for such case

Reply:
Because they were not implemented. Additionally, I did not find any cases using wildcard domains with this modifier in cosmetic rules. This is a good candidate for future improvements.